types/highcharts-vue.d.ts

```diff
@@ -1,43 +1,29 @@
-import _Vue, { VNode, PropOptions, WatchOptionsWithHandler, WatchHandler, CreateElement } from "vue";
-import * as Highcharts from "highcharts"
+// Types for Vue3
+import { DefineComponent, App } from 'vue';
+import * as Highcharts from 'highcharts';
 
-export type ChartUpdateArgs = [boolean, boolean, Highcharts.AnimationOptionsObject]
-
-declare module 'vue/types/options' {
-    interface ComponentOptions<V extends _Vue> {
-        constructorType?: string;
-        options?: Highcharts.Options;
-        callback?: Highcharts.ChartCallbackFunction;
-        updateArgs?: Array<ChartUpdateArgs>;
-        highcharts?: typeof Highcharts;
-        deepCopyOnUpdate?: boolean;
-    }
-}
-
-export interface ChartOptions {
-    tagName?: string;
+export interface ChartProps {
+    constructorType?: string;
+    options: Highcharts.Options;
+    callback?: Highcharts.ChartCallbackFunction;
+    updateArgs?: ChartUpdateArgs;
     highcharts?: typeof Highcharts;
+    deepCopyOnUpdate?: boolean;
 }
 
-export interface ChartPropsObject {
-    constructorType: PropOptions;
-    options: PropOptions;
-    updateArgs: PropOptions;
-    callback: () => void;
+declare module '@vue/runtime-core' {
+    export interface GlobalComponents {
+        [key: string]: DefineComponent<ChartProps, {}, {}>
+    }
 }
 
-export interface ChartWatchObject {
-    options: WatchOptionsWithHandler<any>;
-}
+export type ChartUpdateArgs = [boolean, boolean, Highcharts.AnimationOptionsObject];
```

Comment:
I think we might easily use `Parameters<>` TS utility, and extract the `Chart.update()` function's arguments. Then we should be able to automatically make this type relying on the function itself.